frontend/src/metabase/entities/actions/utils.ts

```diff
@@ -0,0 +1,110 @@
+import _ from "underscore";
+
+import { getDefaultFieldSettings } from "metabase/actions/utils";
+
+import type {
+  ActionFormSettings,
+  FieldType,
+  InputSettingType,
+  Parameter,
+  ParameterType,
+} from "metabase-types/api";
+import type { TemplateTag, TemplateTagType } from "metabase-types/types/Query";
+import type NativeQuery from "metabase-lib/queries/NativeQuery";
+import type Question from "metabase-lib/Question";
+
+import {
+  fieldTypeToParameterTypeMap,
+  dateTypeToParameterTypeMap,
+  fieldTypeToTagTypeMap,
+} from "./constants";
+
+export const removeOrphanSettings = (
+  settings: ActionFormSettings,
+  parameters: Parameter[],
+): ActionFormSettings => {
+  const parameterIds = parameters.map(p => p.id);
+  const fieldIds = Object.keys(settings.fields);
+  const orphanIds = _.difference(fieldIds, parameterIds);
+
+  return {
+    ...settings,
+    fields: _.omit(settings.fields, orphanIds),
```

Comment:
Maybe you could verify this. I think this is equivalent with 
```suggestion
    fields: _.pick(settings.fields, parameterIds),
```

This way we don't need to even declare `fieldIds`

Reply:
Nice! Tried it out and the tests are passing. Thanks @WiNloSt